### DISCUSSION TITLE
[의견] 문맥상 품사가 조금 아쉽네요.

### DISCUSSION THREAD

**Discussion by aajik** [bab2min/kiwipiepy#166] [Q&A] [answered]
Kiwi.version:0.17.1

tokens = kiwi.tokenize(sentence, normalize_coda=True, match_options=Match.ALL | Match.JOIN_NOUN_PREFIX)

text = "여행 가능하다."
[Token(form='여행', tag='NNG', start=0, len=2), Token(form='**가능**', tag='**NNG**', start=3, len=2), Token(form='하', tag='XSA', start=5, len=1), Token(form='다', tag='EF', start=6, len=1), Token(form='.', tag='SF', start=7, len=1)]

여기서 "가능하다"는 형용사로 표현되어야 하는 것이 아닐지요.
"가능"만 보면 명사는 맞지만.... 문맥상으로는 형용사가 되어야 될듯한데....

**Comment by bab2min** [comment]
`가능하다`는 명사 `가능`에 형용사 파생 접미사 `-하다`가 붙어서 만들어진 말이므로 형태소 분석은 위와 같이 되는 게 맞습니다. 다만 저렇게 자세히 분석하는게 아니라 그냥 용언부의 어간만 추출하고 싶으신거라면 형용사 파생 접미사를 결합해주는 `Match.JOIN_ADJ_SUFFIX` 을 사용하시면 됩니다. 마찬가지로 동사에 대해서도 파생 접미사를 결합해주는 `Match.JOIN_VERB_SUFFIX`, 부사에 대해서는 `Match.JOIN_ADV_SUFFIX`가 있으니 필요에 따라 사용하시면 되겠습니다. 자세한 설명은 [여기](https://bab2min.github.io/kiwipiepy/v0.17.0/kr/const.html#kiwipiepy.const.Match.JOIN_ADJ_SUFFIX)에서 살펴보시면 됩니다.

```python
>>> kiwi.tokenize('여행 가능하다', 
    normalize_coda=True, 
    match_options=Match.ALL | Match.JOIN_NOUN_PREFIX | Match.JOIN_ADJ_SUFFIX)

[Token(form='여행', tag='NNG', start=0, len=2), 
 Token(form='가능하', tag='VA', start=3, len=3), 
 Token(form='다', tag='EF', start=6, len=1)]
```

**Comment by aajik** [comment, marked as answer]
네 고맙습니다.

**Comment by aajik** [comment]
새롭게 생성하는 것보다, 관련된 내용은 한곳에서 계속 만들면 좋을 것같아서 여기에...
-->Kiwi.version:0.17.1
text = "몇일 동안 결제 창이 안뜨고, 안되요."
Token(form='안', tag='MAG', start=17, len=1), Token(form='되', tag='VV', start=18, len=1)
"안되요"는 문맥상으로 보면 "동사"로 인식되어야 할 것같은데...

**Reply by bab2min** [reply]
지금까지의 질문들로 보면 형태소라는 개념과 형태소 분석기에 대해서 다르게 이해하고 계신것 같습니다. 위의 문장에서 `안 되다`는 `되다` 앞에 부정을 뜻하는 부사 `안`이 붙은 것입니다. [참고](https://www.joongang.co.kr/article/23701734) 
`안 되다`를 동사로 주장하려면 `안 뜨다`도 동사가 되어야합니다. 형태소 분석기는 되도록 최소의 의미 단위로 분할해주는게 목적이므로 만약 특정 형태소 조합을 하나의 덩어리로 표현하고 싶으시다면 결과를 가지고 필요에 따라 결합하셔서 사용하시거나 사용자 사전에 단어를 등록하시는걸 추천드립니다.

**Reply by aajik** [reply]
네, 그렇군요.
제가 조금 오해를 한 것이 있는 것네요.
Kiwi 생성자에 model_type을 'sbg'를 선택하면 '느리지만 먼 거리에 있는 형태소 간의 관계를 포착할 수 있습니다'라고 설명이 되어 있어서 동사로 인식할 것 같았거던요. 
국립국어원 표준 국어 대사전에 '안-되다'는 동사로 표시가 되어 있고해서....
기존에 활용되고 있는 한국어 형태소 분석기의 대부분은 '안되다'를 '안:부사', '되:동사'로 분류를 하고 있습니다. 트위트만 ('안되다', 'Adjective') 처리하고 있더군요.

**Reply by bab2min** [reply]
@aajik 네 sbg를 사용하면 장거리 의존성을 처리할 수 있는게 맞습니다. 그러나 위의 예시와는 관계가 없습니다. 사전을 보시면 아시겠지만 단일 동사로 쓰이는 '안되다'는 '안 풀리다'와 유사한 의미입니다. 단순히 '되지 않다'라는 뜻이 아니라 '안'과 '되다'가 결합하면서 새로운 의미를 가지게 되었으므로 이를 별개의 동사로 인정하는 겁니다.
위의 예시 문장에서는 일이 잘 안 풀린다는 뜻의 안되다가 아니라 그저 '되지 않는다'는 의미로 '안되다'가 쓰였으므로 `안/MAG 되/VV`로 분석되는게 맞다는 얘기를 드린거였습니다. 비슷한 사례로 `잘못하다`(동사 하나)와 `잘 못하다`(부사와 동사)가 있습니다.
